lib/IRGen/GenDecl.cpp

```diff
@@ -4030,24 +4017,26 @@ IRGenModule::getAddrOfTypeMetadata(CanType concreteType,
 
   auto nominal = concreteType->getAnyNominal();
 
-  llvm::Type *defaultVarTy;
-  unsigned adjustmentIndex;
-
   bool foreign = nominal && requiresForeignTypeMetadata(nominal);
-  bool fullMetadata =
+
+  // Foreign classes and prespecialized generic types do not use an alias into
+  // the full metadata and therefore require a GEP.
+  bool forceGEP =
```

Comment:
I think `fullMetadata` is the better name here, because it describes what we're trying to get, instead of the mechanism by which we get it.

Reply:
Okay, what about `unaliasedFullMetadata`?

The problem with just `fullMetadata` is that to a casual reader of the code, if implies that non-fullMetadata paths don't have "full metadata" when in fact they might. At the very least I wish the comments here were better.